### PR TITLE
Unittests for Star Column Width

### LINKED ISSUE #2129: Broken text wrapping in grid column
```
  <Grid ColumnDefinitions="*, *">
    <TextBlock Grid.Column="1"
               VerticalAlignment="Center"
               Text="He determined to drop his litigation with the monastry, and relinguish his claims to the wood-cuting and fishery rihgts at once. He was the more ready to do this becuase the rights had becom much less valuable, and he had indeed the vaguest idea where the wood and river in quedtion were."
               TextWrapping="Wrap"/>
  </Grid>
```
default

### LINKED ISSUE #2138: Grid layout is broken in some more complex cases with * width
Grid layout is broken in some more complex cases with * width 
examples:

- Path Bounds go outside first column, although path should always fit into first grid column

```XAML
    <Grid ColumnDefinitions="*,*">
        <Path Data="M0,0 L0,10 10,10 10,0Z" Fill="Red" Stroke="Blue" StrokeThickness="5" Stretch="Uniform" Grid.Column="0" 
              HorizontalAlignment="Center" VerticalAlignment="Center" Margin="10" />
        <Border Background="Yellow" BorderBrush="Black" BorderThickness="2" Grid.Column="1" />
    </Grid>
```

- second grid column with static (100 pixels) width is not visible when parent width is decreased during resize

```XAML
    <Grid ColumnDefinitions="*,100" HorizontalAlignment="Left">
        <Path Data="M0,0 L0,10 10,10 10,0Z" Fill="Red" Stroke="Blue" StrokeThickness="5" Stretch="Uniform" Grid.Column="0" 
              HorizontalAlignment="Center" VerticalAlignment="Center" Margin="10" />
        <Border Background="Yellow" BorderBrush="Black" BorderThickness="2" Grid.Column="1" />
    </Grid>
```

related issue #2129

### PR DESCRIPTION
**Pull request by donandren**
Star Column Width don't work well with dynamic desired size based on availablesize

- What does the pull request do?
Unit test for #2138
- What is the current behavior?
measure is performed only once with all available space for the whole grid and resulted desired size can be bigger than actual available space
- What is the updated/expected behavior with this PR?

- How was the solution implemented (if it's not obvious)?

Checklist:

- [x] Added unit tests (if possible)?

### CODE REVIEW

**Comment by grokys** [comment]
@walterlv does this look right to you? Any idea of a fix?

**Comment by grokys** [comment]
Taking a look at these tests I notice 2 things:

- `Layout_LeftAlingment_And_StarColumn_Shouldnt_Stretch` is passing. Should it be failing?
- Comparing `Layout_StarColumn_Should_Pass_Calculated_Width_On_Measure_And_Arrange` with WPF: WPF doesn't measure twice, however it _does_ calculate the fact that due to the columns being `*,*`, 100 pixels is the maximum for each column and passes this to `Measure`

However the outcome is actually the same as WPF: the bounds are correctly set to `0, 50, 100, 100`. So I don't think these tests actually get to the bottom of the problem.

**Comment by donandren** [comment]
#2261 another failing tests for the same grid problem

**Comment by walterlv** [comment]
It seems that this is a fittable idea to solve the dynamic desired size issue. I did this too simple.

File: tests/Avalonia.Controls.UnitTests/GridTests.cs [hunk 1]
```diff
@@ -155,6 +157,68 @@ public void Layout_StarRowColumnWithMaxLength_BoundsCorrect()
...
+                        HorizontalAlignment = HorizontalAlignment.Center,
+                        VerticalAlignment = VerticalAlignment.Center
+                    })
+                }
+            };
+
+            target.Measure(new Size(300, 100));
+            target.Arrange(new Rect(target.DesiredSize));
+
+            Assert.Equal(target.ColumnDefinitions[0].ActualWidth, 100);
```
Comment (walterlv) [inline tests/Avalonia.Controls.UnitTests/GridTests.cs]: The expected argument should be on the left and the actual value should be on the right. So as the below.

**Review by walterlv — CHANGES_REQUESTED** [review]

**Comment by grokys** [comment]
Closing this as we've ported `Grid` from WPF now.

**Pull request closed without merging**